python/tvm/relax/transform/op_legalizer.py

```diff
@@ -0,0 +1,561 @@
+                else:
+                    return a(*a_indices) * b(*b_indices)
+
+            return te.sum(multiply_compute(k), axis=k)
+
+        return te.compute(
+            output_shape,
+            lambda *idx: matmul_compute(*idx),  # pylint: disable=unnecessary-lambda
+            name="matmul",
+        )
+
+    return bb.call_te(te_matmul, call.args[0], call.args[1], primfunc_name_hint="matmul")
+
+
+def _nn_softmax(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.nn.softmax, call.args[0], call.attrs.axis)
+
+
+def _flatten(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.nn.flatten, call.args[0])
+
+
+def _nn_adaptive_max_pool2d(bb: BlockBuilder, call: Call):
+    if call.attrs.out_layout != call.attrs.layout:
+        logging.info(
+            "TOPI adaptive_max_pool2d does not support different input-output "
+            "layouts, and thus cannot be legalized by TOPI"
+        )
+        return call
+
+    if call.attrs.output_size is None:
+        # Todo(ruihang): seems that TOPI adaptive_avg_pool cannot handle the
+        # cases where output_size is not provided
+        return call
+
+    return bb.call_te(
+        topi.nn.adaptive_pool,
+        call.args[0],
+        call.attrs.output_size,
+        pool_type="avg",
+        layout=call.attrs.layout,
+    )
+
+
+# def _nn_softmax_cross_entropy(bb: BlockBuilder, call: Call):
+#     def te_softmax_cross_entropy(x, y):
+#         return _te_cross_entropy(topi.nn.softmax(x), y)
+
+#     return bb.call_te(
+#         te_softmax_cross_entropy, args[0], args[1], primfunc_name_hint="softmax_cross_entropy"
+#     )
+
+
+def _sum(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.sum, call.args[0], call.attrs.axis, call.attrs.keepdims)
+
+
+def _mean(bb: BlockBuilder, call: Call):
+    shape_prod = tvm.tir.const(1, "int32")
+    axis = (
+        call.attrs.axis
+        if call.attrs.axis is not None
+        else range(0, len(call.args[0].struct_info.shape))
+    )
+    for dim in axis:
+        shape_prod = (
+            shape_prod * call.args[0].struct_info.shape[dim if isinstance(dim, int) else dim.value]
+        )
+    sum_var = bb.emit_te(topi.sum, call.args[0], axis, call.attrs.keepdims)
+    return bb.call_te(topi.divide, sum_var, shape_prod)
+
+
+def _image_resize2d(bb: BlockBuilder, call: Call):
+    return bb.call_te(
+        topi.image.resize2d,
+        call.args[0],
+        roi=call.attrs.roi,
+        size=call.attrs.size,
+        layout=call.attrs.layout,
+        method=call.attrs.method,
+        coordinate_transformation_mode=call.attrs.coordinate_transformation_mode,
+        rounding_method=call.attrs.rounding_method,
+        bicubic_alpha=call.attrs.cubic_alpha,
+        bicubic_exclude=call.attrs.cubic_exclude,
+        extrapolation_value=call.attrs.extrapolation_value,
+    )
+
+
+op_legalization_map = {
+    ir.Op.get("relax.nn.conv2d"): _nn_conv2d,
```

Comment:
use a str to op map, then call Op.Get internally, this helps for us to customize